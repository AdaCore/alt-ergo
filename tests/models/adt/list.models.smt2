(set-option :produce-models true)
(set-logic ALL)
(declare-datatype List (
  (empty)
  (cons (head Int) (tail List))
))
(declare-const a List)
(declare-const b List)
(declare-const c List)
(declare-const d List)
(assert (= (tail a) b))
(assert ((_ is cons) c))
(assert (= (tail c) d))
(check-sat)
(get-model)
