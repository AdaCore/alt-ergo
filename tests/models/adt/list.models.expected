
unknown
(
  (define-fun a () List empty)
  (define-fun b () List empty)
  (define-fun c () List (cons 0 empty))
  (define-fun d () List empty)
)
